### ISSUE TITLE
Accidental changes to admin and others users breaks authentication

### ISSUE THREAD

**Issue by flmmartins** [mongodb/mongodb-kubernetes-operator#525]
**What did you do to encounter the bug?**
1. Did some manual changes to admin user via mongo cli

2. Then Mongo tried to recreate the user but couldn't. I got this error:
 mongodb-1:mongodb-agent [2021-06-02T15:14:09.887+0000] [.error] [src/mongoctl/authctl.go:UpsertUserScram:446] [101] <mongodb-1> [15:14:09.887] Error inserting user admin in db  : <mongodb-1> [15:14:09.887] Er │
│ ror executing WithClientFor() for cp=mongodb-1.mongodb-svc.mongodb.svc.cluster.local:27017 (local=false) connectMode=SingleConnect identityUsed=mms-automation@admin[[SCRAM-SHA-256]][20] : multiple write error │
│ s: [{write errors: [{E11000 duplicate key error collection: admin.system.users index: _id_ dup key: { _id: ".admin" }}]}, {<nil>}]

3. Since I cannot login anymore and I can't disable [authentication](https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/docs/users.md ) as mentioned in the README

4. Lost access to my DB

**What did you expect?**
I would expect a way to be able to disable authentication to fix the errors. Or have a solution to force the database to the correct state with the correct users.

**What happened instead?**
I lost access to the database

**Operator Information**
 - Operator Version: 0.6.0
 - MongoDB Image used: 4.4.5

**Kubernetes Cluster Information**
 - Distribution: Azure
 - Version: 1.18.10
 - Image Registry location (quay, or an internal registry): Quay

**Additional context**

I was doing some tests to see if using this Operator was feasible in production. I ran into several issues with users being duplicated because when I was issuing kubectl apply after changing my user roles I notice that were a lot of duplicated entries in system.users. I tried to clean it and end up corrupting the user authentication. 

Additionally every time I changed a user in the mongo cli. The duplicated user errors happened again and again my user was corruption. I expected that if this happens the operator would put my state back but that's not what happens and I always get the error above

This is my final yaml  but during the tests I modified this many times. Generating a lot of duplicated data:

```
apiVersion: mongodbcommunity.mongodb.com/v1
kind: MongoDBCommunity
metadata:
  name: mongodb
  namespace: mongodb
spec:
  members: 2
  type: ReplicaSet
  version: "4.4.5"
  security:
    authentication:
      modes: ["SCRAM"]
  users:
    - name: admin
      passwordSecretRef:
        name: admin-pwd
      roles:
        - name: root 
          db: admin
        - name: root
          db: plugins
      scramCredentialsSecretName: admin
    - name: plugins
      passwordSecretRef:
        name: plugins-pwd
      roles:
        - name: readWrite
          db: plugins
      scramCredentialsSecretName: plugins
  additionalMongodConfig:
    storage.wiredTiger.engineConfig.journalCompressor: zlib
  persistent: true
  statefulSet:
    spec:
      template:
        spec:
          containers:
            - name: mongod
              resources:
                requests:
                  cpu: 1
                  memory: 1Gi
                limits:
                  memory: 8Gi
            - name: mongodb-agent
              resources:
                requests:
                  memory: 50Mi
                limits:
                  cpu: 500m
                  memory: 256Mi
      volumeClaimTemplates:
        - metadata:
            name: data-volume
          spec:
            accessModes: ["ReadWriteOnce"]
            storageClassName: managed-standard-retain
            resources:
              requests:
                storage: 20G
        - metadata:
            name: logs-volume
          spec:
            accessModes: ["ReadWriteOnce"]
            storageClassName: default
            resources:
              requests:
                storage: 1G
```

I have been trying to use this operator for a week now with no success. Is there a way to work around this user corruption?

**Comment by chatton** [comment]
Hi @flmmartins 

Could you provide the contents of of the corresponding config secret that was created by the operator? This should be called `mongodb-config` In your case. In particular I'm interested in the decoded contents of the `auth.usersWanted` array.  

Note: this array will contain credentials in the fields  `scramSha1Creds` and `scramSha256Creds`, the only thing we need to check is the username/db/roles and number of entries.

Once you've saved the state of the above array, could you try configuring `spec.security.authentication.ignoreUnknownUsers` to equal false? This should ensure that any users not listed in the resource definition are removed.

**Comment by flmmartins** [comment]
Thanks for the fast reply!

I added the ignoreUnknown but still when I try to auth with admin against the database:

```
connecting to: mongodb://127.0.0.1:27017/?compressors=disabled&gssapiServiceName=mongodb
Error: Authentication failed. :
connect@src/mongo/shell/mongo.js:374:17
@(connect):2:6
exception: connect failed
exiting with code 1
```

Here is the file:

One interesting thing I noticed is that in one of the iterations I created a root user to the YAML file just to see if I could regain access of the cluster. I removed from the final version but I notice that it's still in the config.


```
{
  "version": 8,
  "processes": [
    {
      "name": "mongodb-0",
      "hostname": "mongodb-0.mongodb-svc.mongodb.svc.cluster.local",
      "args2_6": {
        "net": {
          "port": 27017
        },
        "replication": {
          "replSetName": "mongodb"
        },
        "storage": {
          "dbPath": "/data"
        },
        "storage.wiredTiger.engineConfig.journalCompressor": "zlib",
        "systemLog": {
          "destination": "file",
          "path": "/var/log/mongodb-mms-automation/mongodb.log"
        }
      },
      "featureCompatibilityVersion": "4.4",
      "processType": "mongod",
      "version": "4.4.5",
      "authSchemaVersion": 5
    },
    {
      "name": "mongodb-1",
      "hostname": "mongodb-1.mongodb-svc.mongodb.svc.cluster.local",
      "args2_6": {
        "net": {
          "port": 27017
        },
        "replication": {
          "replSetName": "mongodb"
        },
        "storage": {
          "dbPath": "/data"
        },
        "storage.wiredTiger.engineConfig.journalCompressor": "zlib",
        "systemLog": {
          "destination": "file",
          "path": "/var/log/mongodb-mms-automation/mongodb.log"
        }
      },
      "featureCompatibilityVersion": "4.4",
      "processType": "mongod",
      "version": "4.4.5",
      "authSchemaVersion": 5
    }
  ],
  "replicaSets": [
    {
      "_id": "mongodb",
      "members": [
        {
          "_id": 0,
          "host": "mongodb-0",
          "priority": 1,
          "arbiterOnly": false,
          "votes": 1
        },
        {
          "_id": 1,
          "host": "mongodb-1",
          "priority": 1,
          "arbiterOnly": false,
          "votes": 1
        }
      ],
      "protocolVersion": "1"
    }
  ],
  "auth": {
    "usersWanted": [
      {
        "mechanisms": [],
        "roles": [
          {
            "role": "root",
            "db": "admin"
          },
          {
            "role": "root",
            "db": "plugins"
          }
        ],
        "user": "admin",
        "db": "",
        "authenticationRestrictions": [],
        "scramSha256Creds": {
          "iterationCount": 15000,
        },
        "scramSha1Creds": {
          "iterationCount": 10000,
        }
      },
      {
        "mechanisms": [],
        "roles": [
          {
            "role": "readWrite",
            "db": "plugins"
          }
        ],
        "user": "plugins",
        "db": "",
        "authenticationRestrictions": [],
        "scramSha256Creds": {
          "iterationCount": 15000,
        },
        "scramSha1Creds": {
          "iterationCount": 10000,
        }
      }
    ],
    "disabled": false,
    "authoritativeSet": false,
    "autoAuthMechanisms": [
      "SCRAM-SHA-256"
    ],
    "autoAuthMechanism": "SCRAM-SHA-256",
    "deploymentAuthMechanisms": [
      "SCRAM-SHA-256"
    ],
    "autoUser": "mms-automation",
    "key": "vyRgQRTbjrqCRDE6PyKnLVykRICJm4L4UMC/PDOt3/NTvISxA9+Xw3IpXnZ5Vv2PZANLDVFOt8J8iMWprTu7KW+BTmtBuj+JLALrhl43gWZhXversHwIe5AMjANWRBc3zcr5zPSvs4mTDo2eWjh04iCxKBNh+TaLO/HjfX7M/7Ine0i7ZtwxGB8VZY3Ue7lK1LIMpe3YF4M2DJH1OoofNFf4/gWQtKUMTAWvyMvzkJjpNes9NSm2etf6ThUiAoamaoraKXxwaJ46dbUVolAAy65uQ3Oat/XCMuashq1FlKxt7Ou1A6oR0ibRPWKiqAuZnj8F/q30u8IGz7UDUfUC+/u5eA6yEnp63ka2IF16qscqqw/FmE7dtPQQS37Lpchpfoy3jYBC/eK3n6AneNmR5BqDWW/NVg3JThXU2w6a/bgCJKNMgSIV15peqAoMyyC5AmX7EpYZPI+hgLEdRxdd+1r6ccKCg35YPuOKCGEEu2yrMmLkAgqVzl0i1/LfDva4D0/ARK/eAPQdRAopcYWY05MinjMSqimEssdClduRvWrl1aosQsqTvaYMLng6Y5Y/SuvY6vehrHUmtv2MTawyMiDIXfh4MkmvVeBP3E9SXXcYjczZ+v+aJkHjLQV4F66tsiEm/AStDKPXN8IMkjRONhdRc08=",
    "keyfile": "/var/lib/mongodb-mms-automation/authentication/keyfile",
    "keyfileWindows": "%SystemDrive%\\MMSAutomation\\versions\\keyfile",
    "autoPwd": "removedforsafety"
  },
  "tls": {
    "CAFilePath": "",
    "clientCertificateMode": "OPTIONAL"
  },
  "mongoDbVersions": [
    {
      "name": "4.4.5",
      "builds": [
        {
          "platform": "linux",
          "url": "",
          "gitVersion": "",
          "architecture": "amd64",
          "flavor": "rhel",
          "minOsVersion": "",
          "maxOsVersion": "",
          "modules": []
        },
        {
          "platform": "linux",
          "url": "",
          "gitVersion": "",
          "architecture": "amd64",
          "flavor": "ubuntu",
          "minOsVersion": "",
          "maxOsVersion": "",
          "modules": []
        }
      ]
    }
  ],
  "backupVersions": [],
  "monitoringVersions": [],
  "options": {
    "downloadBase": "/var/lib/mongodb-mms-automation"
  }
}
```

**Comment by rodrigovalin** [comment]
Hello @flmmartins, we are still trying to figure out what's going on in this case. I have a few questions:

1. After applying the changes suggested, in https://github.com/mongodb/mongodb-kubernetes-operator/issues/525#issuecomment-853153533, do you still see the errors reported in your first comment (the one with `Error inserting user admin in db`)
2. From your connection string, what is `gssapiServiceName=mongodb` for?

**Comment by flmmartins** [comment]
Hey Rodrigo,

1. Yes.  I will paste the full log this time. I left the credentials there because I will change them later anyway....
```
 [2021-06-04T09:11:46.885+0000] [.info] [src/director/director.go:tracef:794] <mongodb-1> [09:11:46.885] Running step: 'AdjustUsers' of move 'AdjustUsers'                                │
│ mongodb-1:mongodb-agent [2021-06-04T09:11:46.885+0000] [.info] [src/director/director.go:tracef:794] <mongodb-1> [09:11:46.885] because                                                                          │
│ mongodb-1:mongodb-agent [All the following are true:                                                                                                                                                             │
│ mongodb-1:mongodb-agent     ['currentState.Up' = true]                                                                                                                                                           │
│ mongodb-1:mongodb-agent     ['currentState.IsAuthSpecified' = true]                                                                                                                                              │
│ mongodb-1:mongodb-agent     ['currentState.HasHealthyMajority' = true]                                                                                                                                           │
│ mongodb-1:mongodb-agent     ['desiredState.ProcessType' = mongod]                                                                                                                                                │
│ mongodb-1:mongodb-agent     ['currentState.IsPrimary' = true]                                                                                                                                                    │
│ mongodb-1:mongodb-agent     ['currentState.UsersRight' = false]                                                                                                                                                  │
│ mongodb-1:mongodb-agent     ['currentState.ProcessHasCorrectAutomationCredentials' = true]                                                                                                                       │
│ mongodb-1:mongodb-agent     ['currentState.Version = 4.4.5-ff5cb77101b052fa02da43b8538093486cf9b3f7() is equal to desiredState.Version = 4.4.5-()' = true]                                                       │
│ mongodb-1:mongodb-agent     ['isResponsibleForAdjustingUsers()' = true]                                                                                                                                          │
│ mongodb-1:mongodb-agent     [All the following are true:                                                                                                                                                         │
│ mongodb-1:mongodb-agent         [<current and desired keyfiles contain common key with hash=[254 176 104 16 197 97 194 85 140 11 67 29 81 90 79 88 136 180 198 107 159 140 146 99 166 191 228 189 4 199 19 17]>] │
│ mongodb-1:mongodb-agent     ]                                                                                                                                                                                    │
│ mongodb-1:mongodb-agent ]                                                                                                                                                                                        │
│ mongodb-1:mongodb-agent [2021-06-04T09:11:46.885+0000] [.info] [src/action/adjustusers.go:AdjustUsers:26] <mongodb-1> [09:11:46.885] Running AdjustUsers, processAuthInfo={"UsersWanted":[{"user":"admin","db":" │
│ ","authenticationRestrictions":[],"scramSha1Creds":{"iterationCount":10000,"salt":"czmALafZ2TlLvgd/EYzVYw==","storedKey":"fEBOj11kJ9KJKrK6F8HxEKNKoBQ=","serverKey":"Ysy9/YawdB6MV72ebXrnDshT+SY="},"scramSha256 │
│ Creds":{"iterationCount":15000,"salt":"7awcpw3/WRckoqPEd0Hw/njSYaZhUS7aqJutYw==","storedKey":"pT3qGAAt2SOJOKjuEfVUN0fuOmJ2PtrgP4jl5cgfXck=","serverKey":"d0+tjWvhCrnjk9OgKnKhhMxjQ3j2SEEJXeIznjskn+8="},"roles": │
│ [{"role":"root","db":"admin"},{"role":"root","db":"plugins"}],"inheritedRoles":null,"mechanisms":[],"scope":null},{"user":"plugins","db":"","authenticationRestrictions":[],"scramSha1Creds":{"iterationCount":1 │
│ 0000,"salt":"Zs1IigYhoX7hTHibO2nq1A==","storedKey":"GDkghDDq3BkA/hbohy2j7psbEk0=","serverKey":"AWk0zjewBjSOjsRehyVkY5PGy/w="},"scramSha256Creds":{"iterationCount":15000,"salt":"HKG9W4bkpNQPEkDDv7g/hRTsXC1PNJo │
│ YUFcrcw==","storedKey":"KMP271NJtq09wzurRdbrVsob5JKq7ClWtX91StJC1fI=","serverKey":"r6qTPrcInoFSiY1lSBd/ANy6PHro2GZ6dNBO0F3AbYw="},"roles":[{"role":"readWrite","db":"plugins"}],"inheritedRoles":null,"mechanism │
│ s":[],"scope":null}],"UsersDeleted":null,"DesiredKey":"vyRgQRTbjrqCRDE6PyKnLVykRICJm4L4UMC/PDOt3/NTvISxA9+Xw3IpXnZ5Vv2PZANLDVFOt8J8iMWprTu7KW+BTmtBuj+JLALrhl43gWZhXversHwIe5AMjANWRBc3zcr5zPSvs4mTDo2eWjh04iCxK │
│ BNh+TaLO/HjfX7M/7Ine0i7ZtwxGB8VZY3Ue7lK1LIMpe3YF4M2DJH1OoofNFf4/gWQtKUMTAWvyMvzkJjpNes9NSm2etf6ThUiAoamaoraKXxwaJ46dbUVolAAy65uQ3Oat/XCMuashq1FlKxt7Ou1A6oR0ibRPWKiqAuZnj8F/q30u8IGz7UDUfUC+/u5eA6yEnp63ka2IF16q │
│ scqqw/FmE7dtPQQS37Lpchpfoy3jYBC/eK3n6AneNmR5BqDWW/NVg3JThXU2w6a/bgCJKNMgSIV15peqAoMyyC5AmX7EpYZPI+hgLEdRxdd+1r6ccKCg35YPuOKCGEEu2yrMmLkAgqVzl0i1/LfDva4D0/ARK/eAPQdRAopcYWY05MinjMSqimEssdClduRvWrl1aosQsqTvaYML │
│ ng6Y5Y/SuvY6vehrHUmtv2MTawyMiDIXfh4MkmvVeBP3E9SXXcYjczZ+v+aJkHjLQV4F66tsiEm/AStDKPXN8IMkjRONhdRc08=","DesiredNewKey":"","DesiredKeyHash":"/rBoEMVhwlWMC0MdUVpPWIi0xmufjJJjpr/kvQTHExE=","DesiredNewKeyHash":null │
│ ,"KeyfileHashes":["/rBoEMVhwlWMC0MdUVpPWIi0xmufjJJjpr/kvQTHExE="],"UsingAuth":true}                                                                                                                              │
│ mongodb-1:mongodb-agent [2021-06-04T09:11:46.887+0000] [.info] [src/action/adjustusers.go:adjustUsers:57] <mongodb-1> [09:11:46.887] Upserting user = admin@                                                     │
│ mongodb-1:mongodb-agent [2021-06-04T09:11:46.889+0000] [.error] [src/mongoctl/authctl.go:UpsertUserScram:446] [101] <mongodb-1> [09:11:46.889] Error inserting user admin in db  : <mongodb-1> [09:11:46.889] Er │
│ ror executing WithClientFor() for cp=mongodb-1.mongodb-svc.mongodb.svc.cluster.local:27017 (local=false) connectMode=SingleConnect identityUsed=mms-automation@admin[[SCRAM-SHA-256]][20] : multiple write error │
│ s: [{write errors: [{E11000 duplicate key error collection: admin.system.users index: _id_ dup key: { _id: ".admin" }}]}, {<nil>}]
```

2. To be honest I have no idea. I imagine this comes by default. To login I am doing:

```
$kubectl exec -it mongodb-0 -- bash
Defaulting container name to mongod.
Use 'kubectl describe pod/mongodb-0 -n mongodb' to see all of the containers in this pod.
groups: cannot find name for group ID 2000
I have no name!@mongodb-0:/$ mongo -u admin -p MYPWD
connecting to: mongodb://127.0.0.1:27017/?compressors=disabled&gssapiServiceName=mongodb
Error: Authentication failed. :
connect@src/mongo/shell/mongo.js:374:17
@(connect):2:6
exception: connect failed
exiting with code 1
```

I also tried from another pod. I installed a mongo client and I want to connect to Primary so I do:

```
kubectl -n default exec -it mongodb-new-client -- bash
 I have no name!@mongodb-new-client:/$ mongo --host mongodb/mongodb-svc.mongodb.svc.cluster.local -u admin -p MYPWD
MongoDB shell version v4.2.5
connecting to: mongodb://mongodb-svc.mongodb.svc.cluster.local:27017/?compressors=disabled&gssapiServiceName=mongodb&replicaSet=mongodb
2021-06-04T09:25:11.667+0000 I  NETWORK  [js] Starting new replica set monitor for mongodb/mongodb-svc.mongodb.svc.cluster.local:27017
2021-06-04T09:25:11.667+0000 I  CONNPOOL [ReplicaSetMonitor-TaskExecutor] Connecting to mongodb-svc.mongodb.svc.cluster.local:27017
2021-06-04T09:25:11.687+0000 I  NETWORK  [ReplicaSetMonitor-TaskExecutor] Confirmed replica set for mongodb is mongodb/mongodb-0.mongodb-svc.mongodb.svc.cluster.local:27017,mongodb-1.mongodb-svc.mongodb.svc.cluster.local:27017
2021-06-04T09:25:11.688+0000 I  CONNPOOL [ReplicaSetMonitor-TaskExecutor] Connecting to mongodb-0.mongodb-svc.mongodb.svc.cluster.local:27017
2021-06-04T09:25:11.688+0000 I  CONNPOOL [ReplicaSetMonitor-TaskExecutor] Connecting to mongodb-1.mongodb-svc.mongodb.svc.cluster.local:27017
2021-06-04T09:25:11.694+0000 I  NETWORK  [ReplicaSetMonitor-TaskExecutor] Confirmed replica set for mongodb is mongodb/mongodb-0.mongodb-svc.mongodb.svc.cluster.local:27017,mongodb-1.mongodb-svc.mongodb.svc.cluster.local:27017
2021-06-04T09:25:11.705+0000 I  NETWORK  [js] Marking host mongodb-1.mongodb-svc.mongodb.svc.cluster.local:27017 as failed :: caused by :: Location40659: can't connect to new replica set master [mongodb-1.mongodb-svc.mongodb.svc.cluster.local:27017], err: AuthenticationFailed: Authentication failed.
2021-06-04T09:25:11.706+0000 E  QUERY    [js] Error: can't connect to new replica set master [mongodb-1.mongodb-svc.mongodb.svc.cluster.local:27017], err: AuthenticationFailed: Authentication failed. :
connect@src/mongo/shell/mongo.js:341:17
@(connect):2:6
2021-06-04T09:25:11.707+0000 F  -        [main] exception: connect failed
2021-06-04T09:25:11.707+0000 E  -        [main] exiting with code 1
```

**Comment by chatton** [comment]
Hi @flmmartins, the should correspond to the inverse of `auth.authoritativeSet`in the config file. It looks like maybe there was a problem in the operator and it failed to update this file or perhaps the config value was set at the incorrect level. (`spec.security.authentication.ignoreUnknownUsers`)

Could you double check that this value is set correctly in your MongoDBCommunity resource, and if it is - could you provide the relevant operator logs?

In the meantime I will flag this issue as a feature request for supporting disabled authentication.

**Comment by flmmartins** [comment]
Hello,

So my current mongocommunity is like the first post but @rodrigovalin requested me to add the ignore know users and now this section is like:

```
apiVersion: mongodbcommunity.mongodb.com/v1
kind: MongoDBCommunity
metadata:
  name: mongodb
  namespace: mongodb
spec:
  members: 2
  type: ReplicaSet
  version: "4.4.5"
  security:
    authentication:
      ignoreUnknownUsers: false
      modes: ["SCRAM"]
....
```

The rest is just the same. I don't think this variable is causing the issue because the logs are exactly the same as before adding this variable.

The operator log is as follows and it seems happy:

```
2021-06-08T01:43:50.078Z	DEBUG	controllers/replica_set_controller.go:188	Resetting StatefulSet UpdateStrategy to RollingUpdate	{"ReplicaSet": "mongodb/mongodb"}
2021-06-08T01:43:50.166Z	INFO	controllers/replica_set_controller.go:234	Successfully finished reconciliation	{"ReplicaSet": "mongodb/mongodb", "MongoDB.Spec:": {"members":2,"type":"ReplicaSet","version":"4.4.5","security":{"authentication":{"modes":["SCRAM"],"ignoreUnknownUsers":false},"tls":{"enabled":false,"optional":false,"certificateKeySecretRef":{"name":""},"caConfigMapRef":{"name":""}}},"users":[{"name":"admin","db":"","passwordSecretRef":{"name":"admin-pwd","key":""},"roles":[{"db":"admin","name":"root"},{"db":"plugins","name":"root"}],"scramCredentialsSecretName":"admin"},{"name":"plugins","db":"","passwordSecretRef":{"name":"plugins-pwd","key":""},"roles":[{"db":"plugins","name":"readWrite"}],"scramCredentialsSecretName":"plugins"}],"statefulSet":{"spec":{}},"additionalMongodConfig":{}}, "MongoDB.Status:": {"mongoUri":"mongodb://mongodb-0.mongodb-svc.mongodb.svc.cluster.local:27017,mongodb-1.mongodb-svc.mongodb.svc.cluster.local:27017","phase":"Running","currentStatefulSetReplicas":2,"currentMongoDBMembers":2}}
2021-06-08T01:43:50.166Z	INFO	controllers/replica_set_controller.go:126	Reconciling MongoDB	{"ReplicaSet": "mongodb/mongodb", "MongoDB.Spec": {"members":2,"type":"ReplicaSet","version":"4.4.5","security":{"authentication":{"modes":["SCRAM"],"ignoreUnknownUsers":false},"tls":{"enabled":false,"optional":false,"certificateKeySecretRef":{"name":""},"caConfigMapRef":{"name":""}}},"users":[{"name":"admin","db":"","passwordSecretRef":{"name":"admin-pwd","key":""},"roles":[{"db":"admin","name":"root"},{"db":"plugins","name":"root"}],"scramCredentialsSecretName":"admin"},{"name":"plugins","db":"","passwordSecretRef":{"name":"plugins-pwd","key":""},"roles":[{"db":"plugins","name":"readWrite"}],"scramCredentialsSecretName":"plugins"}],"statefulSet":{"spec":{}},"additionalMongodConfig":{}}, "MongoDB.Status": {"mongoUri":"mongodb://mongodb-0.mongodb-svc.mongodb.svc.cluster.local:27017,mongodb-1.mongodb-svc.mongodb.svc.cluster.local:27017","phase":"Running","currentStatefulSetReplicas":2,"currentMongoDBMembers":2}}
2021-06-08T01:43:50.166Z	DEBUG	controllers/replica_set_controller.go:128	Validating MongoDB.Spec	{"ReplicaSet": "mongodb/mongodb"}
2021-06-08T01:43:50.166Z	DEBUG	controllers/replica_set_controller.go:137	Ensuring the service exists	{"ReplicaSet": "mongodb/mongodb"}
2021-06-08T01:43:50.227Z	INFO	controllers/replica_set_controller.go:382	The service already exists... moving forward: services "mongodb-svc" already exists	{"ReplicaSet": "mongodb/mongodb"}
2021-06-08T01:43:50.227Z	INFO	controllers/replica_set_controller.go:300	Creating/Updating AutomationConfig	{"ReplicaSet": "mongodb/mongodb"}
2021-06-08T01:43:50.251Z	DEBUG	scram/scram.go:160	Credentials have not changed, using credentials stored in: secret/admin-scram-credentials
2021-06-08T01:43:50.271Z	DEBUG	scram/scram.go:160	Credentials have not changed, using credentials stored in: secret/plugins-scram-credentials
2021-06-08T01:43:50.271Z	DEBUG	controllers/replica_set_controller.go:421	AutomationConfigMembersThisReconciliation	{"mdb.AutomationConfigMembersThisReconciliation()": 2}
2021-06-08T01:43:50.271Z	DEBUG	controllers/replica_set_controller.go:323	Waiting for agents to reach version 9	{"ReplicaSet": "mongodb/mongodb"}
2021-06-08T01:43:50.271Z	INFO	agent/agent_readiness.go:52	All 2 Agents have reached Goal state	{"ReplicaSet": "mongodb/mongodb"}
2021-06-08T01:43:50.271Z	INFO	controllers/replica_set_controller.go:271	Creating/Updating StatefulSet	{"ReplicaSet": "mongodb/mongodb"}
2021-06-08T01:43:50.294Z	DEBUG	controllers/replica_set_controller.go:281	Ensuring StatefulSet is ready, with type: RollingUpdate	{"ReplicaSet": "mongodb/mongodb"}
2021-06-08T01:43:50.295Z	INFO	controllers/replica_set_controller.go:286	StatefulSet is ready	{"ReplicaSet": "mongodb/mongodb", "replicas": 2, "generation": 7, "observedGeneration": 7, "updateStrategy": "RollingUpdate"}
2021-06-08T01:43:50.295Z	DEBUG	controllers/replica_set_controller.go:188	Resetting StatefulSet UpdateStrategy to RollingUpdate	{"ReplicaSet": "mongodb/mongodb"}
2021-06-08T01:43:50.371Z	INFO	controllers/replica_set_controller.go:234	Successfully finished reconciliation	{"ReplicaSet": "mongodb/mongodb", "MongoDB.Spec:": {"members":2,"type":"ReplicaSet","version":"4.4.5","security":{"authentication":{"modes":["SCRAM"],"ignoreUnknownUsers":false},"tls":{"enabled":false,"optional":false,"certificateKeySecretRef":{"name":""},"caConfigMapRef":{"name":""}}},"users":[{"name":"admin","db":"","passwordSecretRef":{"name":"admin-pwd","key":""},"roles":[{"db":"admin","name":"root"},{"db":"plugins","name":"root"}],"scramCredentialsSecretName":"admin"},{"name":"plugins","db":"","passwordSecretRef":{"name":"plugins-pwd","key":""},"roles":[{"db":"plugins","name":"readWrite"}],"scramCredentialsSecretName":"plugins"}],"statefulSet":{"spec":{}},"additionalMongodConfig":{}}, "MongoDB.Status:": {"mongoUri":"mongodb://mongodb-0.mongodb-svc.mongodb.svc.cluster.local:27017,mongodb-1.mongodb-svc.mongodb.svc.cluster.local:27017","phase":"Running","currentStatefulSetReplicas":2,"currentMongoDBMembers":2}}
```

**Comment by chatton** [comment]
@flmmartins 

I will try and repo this, do you have some examples of what commands you used through the shell?

Did you add/remove/modify roles etc. ?

**Comment by hilaryming** [comment]
I had the same problem

**Comment by hilaryming** [comment]
So how did you finally solve the problem？@flmmartins

**Comment by flmmartins** [comment]
I couldn't solve the issue and got locked out from the db. I am no longer in the project in question so I cannot provide more logs or configs for this issue.

**Comment by bznein** [comment]
Hello, after some digging this seems to be the same as #805. I see in your yaml you were relying on the default value for the user's db: that was not correctly working and will be fixed in #812

Feel free to reopen this issue if needed
